src/cpp/statistics/fastdds/domain/DomainParticipantImpl.hpp

```diff
@@ -145,6 +147,45 @@ class DomainParticipantImpl : public efd::DomainParticipantImpl
      */
     ReturnCode_t disable_monitor_service() const;
 
+    /**
+     * Fills the in ParticipantProxyData from a MonitorService Message
+     *
+     * @param [out] data Proxy to fill
+     * @param [in] msg MonitorService Message to get the proxy information from.
+     *
+     * @return RETCODE_OK if the operation succeeds.
+     * @return RETCODE_ERROR if the  operation fails.
+     */
+    RTPS_DllAPI ReturnCode_t fill_discovery_data_from_cdr_message(
```

Comment:
This is an internal class. Exporting this symbol (and the ones below) is not needed. You are already exporting the ones in the API.